client/consensus/slots/src/lib.rs

```diff
@@ -288,6 +287,13 @@ pub trait SimpleSlotWorker<B: BlockT> {
 				epoch_data,
 			);
 
+			if let Err(e) = block_import_params {
+				return future::err(e);
+			}
+			let block_import_params = block_import_params.expect(
+				"Potential errors have been checked above; qed"
+			);
```

Comment:
Please, please don't do this!
```suggestion
			let block_import_params = match block_import_params {
				Ok(params) => params,
				Err(e) => return future::err(e),
			};
```